From b3b8d74ac79f540b8fb0d030cec58e80a327f917 Mon Sep 17 00:00:00 2001
From: chir-set <33979252+chir-set@users.noreply.github.com>
Date: Sun, 25 Sep 2022 06:55:15 -0400
Subject: [PATCH 5/6] COMP: Fix build error in
 itkAnisotropicDiffusionVesselEnhancementImageFilter.h
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This commit fixes this build error due to API change in latest ITK version:

31153-SlicerVMTK-git5458f53-g++-64bits-Qt5.15-Release: /.../SlicerVMTK-build/VMTK/vtkVmtk/Segmentation/itkAnisotropicDiffusionVesselEnhancementImageFilter.txx:409:6: error: no matching function for call to ‘itk::AnisotropicDiffusionVesselEnhancementImageFilter<itk::Image<float, 3>, itk::Image<float, 3> >::ResolveTimeStep(std::vector&, std::vector&)’
---
 .../itkAnisotropicDiffusionVesselEnhancementImageFilter.h  | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/vtkVmtk/Segmentation/itkAnisotropicDiffusionVesselEnhancementImageFilter.h b/vtkVmtk/Segmentation/itkAnisotropicDiffusionVesselEnhancementImageFilter.h
index 0e8db5e..b2a894c 100644
--- a/vtkVmtk/Segmentation/itkAnisotropicDiffusionVesselEnhancementImageFilter.h
+++ b/vtkVmtk/Segmentation/itkAnisotropicDiffusionVesselEnhancementImageFilter.h
@@ -228,9 +228,12 @@ private:
     TimeStepType TimeStep;
     std::vector< TimeStepType > TimeStepList;
 #if ITK_VERSION_MAJOR > 5 || (ITK_VERSION_MAJOR == 5 && ITK_VERSION_MINOR >= 3)
-    // ResolveTimeStep argument type was changed to uint_8_t in ITK-5.3.
+    // ResolveTimeStep argument type was changed to uint_8_t in a ITK-5.3 release candidate:
     // https://github.com/InsightSoftwareConsortium/ITK/commit/b66133ab53369bdf265348c0c31cfa6451b53934
-    std::vector< uint8_t > ValidTimeStepList;
+    // Then, ResolveTimeStep argument type was changed to itk::Boolean in a subsequent ITK-5.3 release candidate:
+    // https://github.com/InsightSoftwareConsortium/ITK/commit/bc9ba8540f96c0fa4e9100b25b05eb812074a64e
+    // Therefore, this is the correct implementation for the latest ITK-5.3 version:
+    std::vector< itk::Boolean > ValidTimeStepList;
 #else
     std::vector< bool > ValidTimeStepList;
 #endif
